src/object/cli_obj.c

```diff
@@ -6824,6 +6824,11 @@ dc_obj_punch_task(tse_task_t *task)
 	args = dc_task_get_args(task);
 	D_ASSERTF(args != NULL, "Task Argument OPC does not match DC OPC\n");
 
+	if (unlikely(args->flags & DAOS_COND_PUNCH)) {
+		D_WARN("Ignore condition flag that is not applicable for object.\n");
+		args->flags &= ~DAOS_COND_PUNCH;
+	}
+
```

Comment:
The API does state that the condition punch is only for key:
* \param[in]   flags   Punch flags (currently ignored).  
Could you please change this message to INFO or DEBUG? im afraid warning is a little too much noise in this case.

Reply:
Do we have relative frequent use cases for continual punch object (instead of punch key)? I searched client side logic, only daos_racer does that randomly, no others.

Reply:
right of course we don't because the flag is only a placeholder for now, but that is beside the point. 
IMO warning messages are to users that they have to change their code or adjust something in their usage, where in this case they don't need to since we are just ignoring the flag.
also if someone is debugging a failing test for example and sees this warning, they might say that there is a bug in the test and it needs to change and ignore the actual issues of the failing test.

but the condition here should be changed to:
if (flags != 0)
since this flag in the API is a placeholder and everything is ignored.

Reply:
OK, I will replace D_WANR with D_INF to avoid that.
but "if (args->flags & DAOS_COND_MASK)" is more accurate.

Reply:
OK yes i forgot that you want to clear that bit if it's set, and you require that check.
I meant for other flags that they could be set if they potentially need to be removed. What we could do is just set the entire flag to 0 if any flags are set and add the info message. anyway this is fine for now.